tests/components/application-connector/tools/external-api-mock-app/server.go

```diff
@@ -19,22 +21,54 @@ func main() {
 	}
 	log.SetLevel(logLevel)
 
-	log.Infof("Starting frog auth server application")
+	log.Infof("Starting mock application")
 	log.Infof("Config: %s", cfg.String())
 
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+
 	basicAuthCredentials := test_api.BasicAuthCredentials{User: cfg.BasicAuthUser, Password: cfg.BasicAuthPassword}
 	oAuthCredentials := test_api.OAuthCredentials{ClientID: cfg.OAuthClientID, ClientSecret: cfg.OAuthClientSecret}
-	expectedRequestParameters := test_api.ExpectedRequestParameters{Headers: cfg.RequestHeaders, QueryParameters: cfg.RequestQueryParameters}
+	tokens := make(map[string]test_api.OAuthToken)
+
+	go func() {
+		expectedRequestParameters := test_api.ExpectedRequestParameters{Headers: cfg.RequestHeaders, QueryParameters: cfg.RequestQueryParameters}
```

Comment:
Maybe we can move it higher? Just to leave the `go func() {` blocks just to initialize servers. Just like `basicAuthCredentials` and so on